lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2749,6 +2749,7 @@ static bool ParseSILArgs(SILOptions &Opts, ArgList &Args,
                    Opts.EnableLifetimeDependenceDiagnostics);
 
   Opts.VerifyAll |= Args.hasArg(OPT_sil_verify_all);
+  Opts.VerifyAll |= CONDITIONAL_ASSERT_enabled();
```

Comment:
I don't think it's a good idea to enable VerifyAll with `-compiler-assertions`, because it increases compile time significantly (it runs the verifier after each pass).
Instead it would be better that `-compiler-assertions` enables verification at all. So that it behaves the same way as a compiler built with asserts.
That can be done in `verificationEnabled`: https://github.com/swiftlang/swift/blob/b58a908082692bd84b3f52064e77065840e24112/lib/SIL/Verifier/SILVerifier.cpp#L7324

Reply:
That looks like a much better approach!  Changed.